system/jlib/jfile.cpp

```diff
@@ -749,74 +749,94 @@ bool CFile::remove()
 #endif
 }
 
-static std::atomic<unsigned> renameRetries{(unsigned)-1}; // initially uninitialized, set on 1st use (in rename)
-
-void setRenameRetries(unsigned numRetries)
+static std::atomic<unsigned> renameRetries = (unsigned)-1; // initially uninitialized, set on 1st use (in rename)
+static std::atomic<bool> manualRenameChk = false;
+void setRenameRetries(unsigned _renameRetries, bool _manualRenameChk)
 {
-    renameRetries = numRetries;
+    manualRenameChk = _manualRenameChk;
+    renameRetries = _renameRetries;
 }
 
-static unsigned getRenameRetries()
+static void initRenameRetrySettings()
 {
-    unsigned retries;
+    if ((unsigned)-1 != renameRetries)
+        return;
     // NB: potentially could be >1 thread here, but that's ok.
     try
     {
-        retries = getGlobalConfigSP()->getPropInt("expert/@numRenameRetries");
+        Owned<IPropertyTree> globalConfig = getGlobalConfigSP();
+        Owned<IPropertyTree> config = getComponentConfigSP();
+        manualRenameChk = globalConfig->getPropBool("expert/@manualRenameChk", config->getPropBool("expert/@manualRenameChk"));
```

Comment:
The calls are in the wrong order.  This means the global config will take precedence over the component config.

Reply:
yes, agreed, will flip.